From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH] lightnvm: fix warning: ret may be used uninitialized
Date: Mon, 4 Apr 2016 14:05:51 -0400
References: FATE#319466
Patch-mainline: v4.7-rc1
Git-commit: 57aac2f1be4a0711b7f24f7d367e8672ebaa3844

This fixes the following warnings:
drivers/lightnvm/sysblk.c:125:9: warning: ‘ret’ may be used
uninitialized in this function

drivers/lightnvm/sysblk.c:275:15: warning: ‘ret’ may be used
uninitialized in this function

In both cases, ret is only set from within a loop that may not be entered.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
Reviewed-by: Johannes Thumshirn <jthumshirn@suse.de>
---

 drivers/lightnvm/sysblk.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/lightnvm/sysblk.c
+++ b/drivers/lightnvm/sysblk.c
@@ -122,7 +122,7 @@ static int nvm_get_all_sysblks(struct nv
 				struct ppa_addr *ppas, nvm_bb_update_fn *fn)
 {
 	struct ppa_addr dppa;
-	int i, ret;
+	int i, ret = 0;
 
 	s->nr_ppas = 0;
 
@@ -272,7 +272,7 @@ static int nvm_write_and_verify(struct n
 {
 	struct nvm_system_block nvmsb;
 	void *buf;
-	int i, sect, ret, bufsz;
+	int i, sect, ret = 0, bufsz;
 	struct ppa_addr *ppas;
 
 	nvm_cpu_to_sysblk(&nvmsb, info);



